github-actions.md

```diff
@@ -0,0 +1,30 @@
+# GitHub Actions
+
+Github actions should be used for Continuous Integration. Actions are typically used to enforce checks before merging branches into main (such as formatting, linting and testing), as well as to publish libraries or [deploy](continuous-deployment.md) build artifacts (via Riff-Raff).
+
+## Specifying versions for actions you use
+
+Github Actions workflows can invoke other actions via `uses` steps. When specifying the version to use, one can typically:
+
+- Reference via a Git tag e.g `uses: actions/checkout@v4`. These correspond to tagged releases on Github - see [actions/checkout/releases](https://github.com/actions/checkout/releases) for examples.
+- Reference via a Git Commit SHA e.g. `uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1`.
+
+**We recommend specifying the version of Github Actions workflows as a commit SHA**.
+
+A security issue arises when using tags to specify versions: code executed by the underlying action can be changed without creating a new release. A malicious actor could exploit this in order to run code within our own workflows. See this [blog post](https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash) for a complete example.
```

Comment:
Is it worth also noting this recommendation differs from [GitHub's](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions)?

IMO GitHub's recommendation strikes a better balance between DX and security, though admittedly is more difficult to enforce (once we get there!).

Reply:
GitHub seem to also recommend pinning on the commit digest?

Reply:
I was referring to the "Pin actions to a tag only if you trust the creator" section. I'd suggest we're able to trust actions authored by ourselves, GitHub, AWS, etc. Actions from third-party individuals, such as [EnricoMi/publish-unit-test-result-action](https://github.com/guardian/content-api/blob/4cb34b2ee185203ac5ca059c0d08615ed2c8a8c3/.github/workflows/porter.yml#L71) would be pinned.

Reply:
Personally I think the principle of zero-trust applies here. Sure, I trust no one at the Guardian would maliciously attempt to steal my repo secrets, but if there's a compromised GitHub action repo then I'd prefer if that didn't cascade to my secrets being stolen.

If dependabot is able to flag updates with commit digests then I think we should be able to apply updates with a small amount of toil (I generally don't want random, unrequested updates landing in my codebases regardless of the security implications).

Reply:
> If dependabot is able to flag updates with commit digests then I think we should be able to apply updates with a small amount of toil

Agree in principle. However in practice, I think the story is a little different as:
- Usage of Dependabot to bump GHA isn't universally applied
- Timely review of Dependabot PRs is also not universal; there are a couple thousand [PRs open today](https://github.com/pulls?q=is%3Aopen+is%3Apr+org%3Aguardian+author%3Aapp%2Fdependabot)

The volume of open PRs is probably a signal of some missing tooling or processes TBH, but that's a different problem to solve...

Reply:
Thanks for the feedback both. I'll leave the recommendation as it is, but mention that we're taking a slightly different stance to the one that is recommended by Github.